src/devices/machine/jvsdev.cpp

```diff
@@ -82,11 +82,24 @@ void jvs_device::message(uint8_t dest, const uint8_t *send_buffer, uint32_t send
 		next_device->message(dest, send_buffer, send_size, recv_buffer, recv_size);
 }
 
+int jvs_device::device_handle_message(const uint8_t *send_buffer, uint32_t send_size, uint8_t *&recv_buffer)
+{
+	// Override this function to provide device-specific JVS message handling.
+	// If -1 is returned then the base message handler will be used.
+	return -1;
+}
```

Comment:
This is an antipattern.  You should provide default message handling in a protected virtual method, and allow derived classes to override it, and call the base method if they don’t handle the message specially.